src/amqp-connector.js

```diff
@@ -7,20 +7,19 @@ class AmqpConnector {
   }
 
   async connect() {
-    try {
-      this.connection = await amqp.connect('amqp://' + this.host);
-      logger.info('RabbitMQ connected');
-      process.once('SIGINT', () => {
-        this.connection.close();
-      });
-      this.channel = await this.connection.createChannel();
-      logger.info('RabbitMQ channel ready');
-    } catch (error) {
-      logger.warn(error);
-      logger.log('RabbitMQ connection failed. Reconnecting in 1 s ...');
-      setTimeout(() => {
-        this.connect();
-      }, 1000);
+    while(!this.connection) {
+      try {
+        this.connection = await amqp.connect('amqp://' + this.host);
+        logger.info('RabbitMQ connected');
+        process.once('SIGINT', () => {
+          this.connection.close();
+        });
+        this.channel = await this.connection.createChannel();
+        logger.info('RabbitMQ channel ready');
+      } catch (error) {
+        logger.warn(error);
+        logger.log('RabbitMQ connection failed. Reconnecting ...');
+      }
```

Comment:
While this solution looks better, it is now hammering the host with a bunch of requests, because there is no longer a `setTimeout()`.
I'm still not quite sure whether this is a good solution at all. It should not be the responsibility of the Node.js app to wait for other services. I'd much rather like to see a solution like [this](https://stackoverflow.com/questions/31746182/docker-compose-wait-for-container-x-before-starting-y/41854997#41854997), or, probably even better, include something like [this](https://github.com/vishnubob/wait-for-it#wait-for-it) in the `Dockerfile`s.

Reply:
It is only an edge case that RabbitMQ is not available when the app starts. Normally it is already available. So I think this should not be a problem for now.
To reduce the number of requests, we could add busy waiting there.